docs/book/codegen.md

```diff
@@ -5,14 +5,16 @@ generators to create optimized code for production. These generators will
 inspect the provided classes, resolve their dependencies, and generate factories
 based on these results.
 
-> ### Requirements
+> ### Requirements (superseded)
 >
-> This feature requires [zend-code](https://docs.zendframework.com/zend-code/),
+> Before version 3.1, this feature required [zend-code](https://docs.zendframework.com/zend-code/),
 > which you can add to your project using Composer:
 >
 > ```bash
-> $ composer require zendframework/zend-code
+> $ composer require --dev zendframework/zend-code
 > ```
+>
+> **Since version 3.1 and up, this is no longer required.**
 
```

Comment:
I'm unsure if we could move this block all the way down to the bottom and make it a regular h2? Thoughts?